### PR TITLE
Set the last edited map file as a default for the new Standard game

### PR DESCRIPTION
**Pull request by Districh-ru**
This PR sets the last loaded or saved map in Editor as a default selected map for the new Standard Game. It will be selected only if the map can be played.

### CODE REVIEW

File: src/fheroes2/editor/editor_interface.cpp [hunk 1]
```diff
@@ -1607,6 +1608,11 @@ namespace Interface
 
         _loadedFileName = System::truncateFileExtensionAndPath( filePath );
 
+        // Set the loaded map as a default map for the new Standard Game.
+        Maps::FileInfo & mapInfo = Settings::Get().getCurrentMapInfo();
```
Comment (ihhub) [inline src/fheroes2/editor/editor_interface.cpp:1612]: This is definitely a hack. We should use `SetCurrentFileInfo()` method for this. Create **FileInfo** object and use **readResurrectionMap()** method for this matter. Going forward: you can create a function in this file which accepts `Maps::Map_Format::BaseMapFormat`, create a new method in **Maps::FileInfo** called for example `loadResurrectionMap()` and move most of the logic there from `Maps::FileInfo::readResurrectionMap()`.

By the way, line 396 in **maps_fileinfo.cpp** should be `Maps::Map_Format::BaseMapFormat map;` as we don't need the full map to be loaded.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @Districh-ru , I left a suggestion in this pull request. Would you mind to take a look when you have spare time?

File: src/fheroes2/editor/editor_interface.cpp (hunk 1 above)
Comment (Districh-ru) [reply src/fheroes2/editor/editor_interface.cpp]: @ihhub, many thanks for the suggestion!

**Review by ihhub — APPROVED** [review]

**Review by Branikolog — APPROVED** [review]

**Comment by ihhub** [comment]
@Districh-ru , huge thanks for this update!

**Pull request merged**